datafusion/sqllogictest/test_files/unnest.slt

```diff
@@ -860,6 +860,12 @@ select count(*) from (select unnest(range(0, 100000)) id) t inner join (select u
 ----
 100000
 
+# Implicit LATERAL support
+# Issue: https://github.com/apache/datafusion/issues/13659
+# TODO: https://github.com/apache/datafusion/issues/10048
+query error DataFusion error: This feature is not implemented: Physical plan does not support logical expression OuterReferenceColumn\(List\(Field \{ name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: \{\} \}\), Column \{ relation: Some\(Bare \{ table: "u" \}\), name: "column1" \}\)
+select * from unnest_table u, unnest(u.column1);
```

Comment:
I think it's better to add a logical plan test like
```
explain select * from unnest_table u, unnest(u.column1);
```
You can refer to how other `LATERAL` tests do it.

Reply:
Added!

Reply:
It's good to keep this one and another planning one in the sqllogictest (for end-to-end SQL test).

Just like the following case:
https://github.com/apache/datafusion/blob/d3cfc45cef28a2d707bdd4b14da33a4d38a209d6/datafusion/sqllogictest/test_files/joins.slt#L4051-L4053

and

https://github.com/apache/datafusion/blob/d3cfc45cef28a2d707bdd4b14da33a4d38a209d6/datafusion/sqllogictest/test_files/joins.slt#L4068-L4072